static/js/chart.js

```diff
@@ -0,0 +1,398 @@
+        percentText.setAttribute('text-anchor', 'middle')
+        percentText.setAttribute('fill', type === 'source_remaining' ? '#96A6FF' : '#f59e0b')
+        percentText.setAttribute('font-family', 'Arial, sans-serif')
+        percentText.setAttribute('font-size', '32px')
+        percentText.setAttribute('font-weight', 'bold')
+        percentText.textContent = `${percentage}%`
+        
+        svg.appendChild(percentText)
+      }
+    },
+    
+    drawFallbackWalletIcon(svg, x, y, type, percentage, targetName = null) {
+      // Fallback wallet icon when PNG fails to load
+      const rect = document.createElementNS('http://www.w3.org/2000/svg', 'rect')
+      rect.setAttribute('x', x - 30)
+      rect.setAttribute('y', y - 30)
+      rect.setAttribute('width', 60)
+      rect.setAttribute('height', 60)
+      rect.setAttribute('rx', 12)
+      rect.setAttribute('fill', (type === 'source' || type === 'source_remaining') ? (type === 'source_remaining' ? '#96A6FF' : '#6366f1') : '#f59e0b')
+      rect.setAttribute('stroke', '#1f2937')
+      rect.setAttribute('stroke-width', 2)
+      
+      svg.appendChild(rect)
+      
+      // Add Bitcoin symbol
+      const text = document.createElementNS('http://www.w3.org/2000/svg', 'text')
+      text.setAttribute('x', x)
+      text.setAttribute('y', y + 5)
+      text.setAttribute('text-anchor', 'middle')
+      text.setAttribute('fill', 'white')
+      text.setAttribute('font-family', 'Arial, sans-serif')
+      text.setAttribute('font-size', '24')
+      text.setAttribute('font-weight', 'bold')
+      text.textContent = '₿'
+      
+      svg.appendChild(text)
+      
+      // Add name and percentage below icon for targets and source_remaining
+      if (type === 'target' || type === 'source_remaining') {
+        // Add name text
+        if (targetName) {
+          const nameText = document.createElementNS('http://www.w3.org/2000/svg', 'text')
+          nameText.setAttribute('x', x)
+          nameText.setAttribute('y', y + 45)
+          nameText.setAttribute('text-anchor', 'middle')
+          nameText.setAttribute('fill', type === 'source_remaining' ? '#96A6FF' : '#374151')
+          nameText.setAttribute('font-family', 'Arial, sans-serif')
+          nameText.setAttribute('font-size', '14px')
+          nameText.setAttribute('font-weight', 'bold')
+          nameText.textContent = targetName
+          
+          svg.appendChild(nameText)
+        }
+        
+        // Add percentage text below name
+        const percentText = document.createElementNS('http://www.w3.org/2000/svg', 'text')
+        percentText.setAttribute('x', x)
+        percentText.setAttribute('y', y + 65)
+        percentText.setAttribute('text-anchor', 'middle')
+        percentText.setAttribute('fill', type === 'source_remaining' ? '#96A6FF' : '#f59e0b')
+        percentText.setAttribute('font-family', 'Arial, sans-serif')
+        percentText.setAttribute('font-size', '16px')
```

Comment:
can we get rid of font styles inside the svg? and maybe let the be picked up from the style?